core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -878,7 +886,7 @@ public Object updateEvent() {
 
   @SuppressWarnings("checkstyle:CyclomaticComplexity")
   private void cleanUncommittedAppends(Set<ManifestFile> committed) {
-    if (cachedNewDataManifests != null) {
+    if (!cachedNewDataManifests.isEmpty()) {
```

Comment:
Any reason we need to change this? I see you covered all the cases in this class where we do explicit null checks and replaced them with empty checks so I think logically we're good. More so just curious if it's required for this change (also I think nulling it out would be quicker than clearing, since clear goes through very element and nulls it out but that's probably minor).

Reply:
I see, we're doing the same thing for delete manifests as well. Should be fine. For context, I was particularly scrutinizing at this because this path was recently added as part of the 1.4.3 patch release to fix an issue where we were skipping added files as part of transaction retries: https://github.com/apache/iceberg/pull/9230

Reply:
Yea, the reason for this change was just to be consistent with the delete-files approach.